compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1086,7 +1094,34 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             *region,
                             ty::TypeAndMut { ty: *t_type, mutbl: mutability.invert() },
                         );
-                        err.note(&format!("you need `{}` instead of `{}`", trait_type, rcvr_ty));
+                        let msg = format!("you need `{}` instead of `{}`", trait_type, rcvr_ty);
+                        let mut kind = &self_expr.kind;
+                        while let hir::ExprKind::AddrOf(_, _, expr)
+                        | hir::ExprKind::Unary(hir::UnOp::Deref, expr) = kind
+                        {
+                            kind = &expr.kind;
+                        }
+                        if let hir::ExprKind::Path(hir::QPath::Resolved(None, path)) = kind
+                            && let hir::def::Res::Local(hir_id) = path.res
+                            && let Some(hir::Node::Pat(binding)) = self.tcx.hir().find(hir_id)
+                            && let parent_hir_id = self.tcx.hir().get_parent_node(binding.hir_id)
+                            && let Some(hir::Node::Param(param)) = self.tcx.hir().find(parent_hir_id)
+                            && let parent_hir_id = self.tcx.hir().get_parent_node(param.hir_id)
+                            && let Some(node) = self.tcx.hir().find(parent_hir_id)
```

Comment:
That's a nice example of code that will be simplified with #106403, we should probably wait for it

Reply:
You need to rebase to use the new HIR methods here

Reply:
fixed